R/tokenization.R

```diff
@@ -13,768 +13,140 @@
-              "wordpiece_tokenizer" = wordpiece_tokenizer)
-  class(obj) <- "FullTokenizer"
-  return(obj)
-}
-
-
-
-# generic tokenize --------------------------------------------------------
-
-
-#' Tokenizers for various objects.
-#'
-#' This tokenizer performs some basic cleaning, then splits up text on
-#' whitespace and punctuation.
-#'
-#' @param tokenizer The Tokenizer object to refer to.
-#' @param text The text to tokenize. For tokenize.WordpieceTokenizer, the text
-#'   should have already been passed through BasicTokenizer.
-#'
-#' @return A list of tokens.
-#' @export
-tokenize <- function (tokenizer, text) {
-  UseMethod("tokenize", tokenizer)
-}
-
-
-# tokenize.FullTokenizer --------------------------------------------------
-
-
-#' @describeIn tokenize Tokenizer method for objects of FullTokenizer class.
-#' @export
-#'
-#' @examples
-#' \dontrun{
-#' tokenizer <- FullTokenizer("vocab.txt", TRUE)
-#' tokenize(tokenizer, text = "a bunch of words")
-#' }
-tokenize.FullTokenizer <- function(tokenizer, text) {
-  b_tokens <- tokenize(tokenizer$basic_tokenizer, text) # this is really ugly.
-
-  # We can't use purrr::map_chr here, since the output of .f is itself a vector
-  # of variable length (map_chr died trying...). Use map + unlist.
-  split_tokens <- purrr::map(b_tokens,
-                             function(bt) {
-                               tokenize(tokenizer$wordpiece_tokenizer, bt)
-                             })
-  return(unlist(split_tokens))
-}
-
-
-
-# class BasicTokenizer ----------------------------------------------------
-
-
-#' Construct objects of BasicTokenizer class.
-#'
-#' (I'm not sure that this object-based approach is best for R implementation,
-#' but for now just trying to reproduce python functionality.)
-#'
-#' Has methods: `tokenize.BasicTokenizer()` `run_strip_accents.BasicTokenizer()`
-#' (internal use) `run_split_on_punc.BasicTokenizer()` (internal use)
-#' `tokenize_chinese_chars.BasicTokenizer()` (internal use)
-#' `is_chinese_char.BasicTokenizer()` (internal use)
-#' `clean_text.BasicTokenizer()` (internal use)
-#'
-#' @param do_lower_case Logical; the value to give to the "do_lower_case"
-#'   argument in the BasicTokenizer object.
+#' @param seq_list Character vector or list; text to tokenize.
+#' @param vocab A wordpiece vocabulary.
+#' @param pad_to_length Integer; optional length to pad sequences to.
+#' @param sep_token Character; token to use at end of each segment. Should exist
+#'   in \code{vocab}.
+#' @param cls_token Character; token to use at start of sequence. Should exist
+#'   in \code{vocab}.
+#' @param pad_token Character; token to use for padding sequence. Should exist
+#'   in \code{vocab}.
 #'
-#' @return an object of class BasicTokenizer
+#' @return The tokenized output as a named integer vector, with token type ids
+#'   attached as an attribute.
 #' @export
-#'
 #' @examples
 #' \dontrun{
-#' b_tokenizer <- BasicTokenizer(TRUE)
+#' # assuming vocabulary `vocab`
```

Comment:
Can we update this to use the built-in {wordpiece} vocab, so it doesn't have to be dontrun?